resample/bootstrap.py

```diff
@@ -372,3 +372,11 @@ def _confidence_interval_bca(
 
     quant = quantile_function_gen(thetas)
     return quant(p_low), quant(p_high)
+
+
+del Callable
+del Generator
+del Optional
+del Tuple
+del Union
+del Sequence
```

Comment:
Rather than importing and then deleting, does it make sense to import within an `if TYPE_CHECKING` conditional? Then I believe none of these should be exposed as part of the API, which I think is what we are trying to avoid here.

https://docs.python.org/3/library/typing.html?highlight=type_checking#typing.TYPE_CHECKING

Reply:
I assume the runtime cost from importing the typing module is negligible, so it is not necessary to make this conditional. Have you tried this, because I am not sure it work at all with such an if? I have seen this approach of deleting to clean up the module namespace elsewhere, so this is something people do.